crates/red_knot_project/tests/check.rs

```diff
@@ -84,6 +87,10 @@ fn run_corpus_tests(pattern: &str) -> anyhow::Result<()> {
 
     let mut db = setup_db(&root, system.clone())?;
 
+    Program::get(&db)
+        .set_python_version(&mut db)
+        .to(PythonVersion::latest());
```

Comment:
This is necessary because some of the parser and linter corpus files use e.g. `except *` syntax, for which we need the `BaseExceptionGroup` symbol to be available, and it's only available on Python 3.11+. We would have had to make this change anyway pretty soon, since we'll probably be integrating @ntBre's work on improved syntax-error detection into red-knot at some point in the next few weeks.

Reply:
I don't think this change is necessary for the syntax-work integration because this test only asserts that we can pull all types. It doesn't assert the absence of diagnostics. 

It's not clear to me why this change is necessary for what you did in this PR. Does that mean that Red Knot now panics if someone uses `except *` but sets `python-version=3.9` in their project configuration?

Reply:
> I don't think this change is necessary for the syntax-work integration because this test only asserts that we can pull all types. It doesn't assert the absence of diagnostics.

Hmm, fair point

> Does that mean that Red Knot now panics if someone uses `except *` but sets `python-version=3.9` in their project configuration?

Only if the `test` or `debug_assertions` features is enabled for their red-knot build. Which shouldn't ever really be the case for end users of red-knot.

Reply:
Yeah, not having any test coverage that red knot indeed doesn't panic in those cases those make me a little nervous and in favor of removing the panics.